### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Setup unit tests

## Changes

- Added unit tests
- Small changes in `src/` to accommodate tests.
- fixed `lengthOfNumber(0) = 0`
(There is also coverage included in the setup, but it doesn't work currently)

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,25 @@ make emulator
 ```
 The file will be available at `bin/app.elf`.
 
+## Testing
+
+### Running unit tests
+There are tests on some of the functions, which doen't rely on Ledger specfic behaviour.
```
Comment (orhoj) [inline README.md:71]: doen't

File: src/numberHelpers.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+
```
Comment (orhoj) [inline src/numberHelpers.h:1]: Add:
```
#ifndef _NUMBER_HELPERS_H
#ifndef _NUMBER_HELPERS_H
...
...
...
#endif
```

File: src/base58check.c [hunk 3]
```diff
@@ -20,7 +20,12 @@
 ********************************************************************************/
 
 #include <stdlib.h>
+#if TEST
```
Comment (orhoj) [inline src/base58check.c:23]: Please document why this is done.

File: unit_tests/gen_coverage.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (orhoj) [inline unit_tests/gen_coverage.sh:1]: Do we need this file if we are not going to use the coverage information for now?

File: unit_tests/tests/testNumberHelpers.c [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+#include <string.h>
+
+#include <stdint.h>  // uint*_t
+#include <stddef.h>  // size_t
+
+#include <cmocka.h>
+
+#include "numberHelpers.c"
+
+static void test_lengthOfNumbers() {
```
Comment (orhoj) [inline unit_tests/tests/testNumberHelpers.c:15]: It would be good to include edge cases, i.e. have tests for the maximum available input.

File: unit_tests/tests/testNumberHelpers.c [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+static void test_numberToText_longer_number() {
+    uint8_t text[4];
+    // This should fail, because the text is not long enough
+    int result = numberToText(text, 12041);
+    assert_int_equal(result, -1);
+}
+*/
+static void test_bin2dec() {
+    uint8_t text[4];
+    bin2dec(text, 2001);
```
Comment (orhoj) [inline unit_tests/tests/testNumberHelpers.c:63]: Since we have a string termination character when using `bin2dec`, this should not go well.

File: README.md (hunk 1 above)
Comment (orhoj) [reply README.md]: There are unit tests on some of the functions that do not rely on Ledger specific libraries.
